submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/SubmitProjectPermissionTest.kt

```diff
@@ -0,0 +1,77 @@
+import ac.uk.ebi.biostd.client.integration.web.SecurityWebClient
+import ac.uk.ebi.biostd.common.config.PersistenceConfig
+import ac.uk.ebi.biostd.common.config.SubmitterConfig
+import ac.uk.ebi.biostd.itest.common.BaseIntegrationTest
+import ac.uk.ebi.biostd.itest.common.TestConfig
+import ac.uk.ebi.biostd.itest.entities.RegularUser
+import ac.uk.ebi.biostd.itest.entities.SuperUser
+import ebi.ac.uk.dsl.line
+import ebi.ac.uk.dsl.tsv
+import ebi.ac.uk.test.createFile
+import io.github.glytching.junit.extension.folder.TemporaryFolder
+import io.github.glytching.junit.extension.folder.TemporaryFolderExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.web.server.LocalServerPort
+import org.springframework.context.annotation.Import
+import org.springframework.http.HttpStatus
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.web.client.HttpServerErrorException
+
+@ExtendWith(TemporaryFolderExtension::class)
+internal class SubmitProjectPermissionTest(private val tempFolder: TemporaryFolder) : BaseIntegrationTest(tempFolder) {
+    @Nested
+    @ExtendWith(SpringExtension::class)
+    @Import(value = [TestConfig::class, SubmitterConfig::class, PersistenceConfig::class, TestConfig::class])
+    @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+    @DirtiesContext
+    inner class SubmitTest {
+        @LocalServerPort
+        private var serverPort: Int = 0
+
+        private val project = tsv {
+            line("Submission", "AProject")
+            line()
+
+            line("Project")
+        }
+
+        private val projectFile = tempFolder.createFile("a-project.tsv", project.toString())
+
+        private lateinit var superUserWebClient: BioWebClient
+        private lateinit var regularUserWebClient: BioWebClient
+
+        @BeforeAll
+        fun init() {
+            val securityClient = SecurityWebClient.create("http://localhost:$serverPort")
+            securityClient.registerUser(SuperUser.asRegisterRequest())
+            securityClient.registerUser(RegularUser.asRegisterRequest())
+
+            superUserWebClient = securityClient.getAuthenticatedClient(SuperUser.email, SuperUser.password)
+            regularUserWebClient = securityClient.getAuthenticatedClient(RegularUser.email, RegularUser.password)
+        }
+
+        @Test
+        fun `create project with superuser`() {
+            val response = superUserWebClient.submitProject(projectFile)
+            assertThat(response).isNotNull
+            assertThat(response.statusCode).isEqualTo(HttpStatus.OK)
```

Comment:
what is the difference between this test and ```fun `submit project`()```

Reply:
The idea here is that in this test there're two clients. The superuser and regular user, so it's like the "happy path" test for superuser creating a project from the security point of view.

Reply:
but in `submit project` test are we submitting with superuser too. arent we?